pfs_middleware/pfs_middleware/middleware.py

```diff
@@ -1894,7 +1894,9 @@ def coalesce_object(self, ctx, auth_cb):
                 # Gotta check auth for all of the segments, too
                 bimodal_checker = ctx.req.environ[utils.ENV_BIMODAL_CHECKER]
                 acl_env = ctx.req.environ.copy()
-                acl_env['PATH_INFO'] = elem_container_path
+                acl_env['PATH_INFO'] = swift_code.bytes_to_wsgi(
+                    elem_container_path.encode(
+                        'utf8', errors='surrogateescape'))
```

Comment:
Oh, yeah -- `errors='surrogateescape'` shouldn't be necessary -- the response coming out of proxyfs shouldn't have any surrogates.

Reply:
Looks like it shouldn't matter as long as there are no errors, though:

```
>>> u'\N{SNOWMAN}'.encode('utf8', errors='surrogateescape').decode('utf8', errors='surrogateescape')
u'\u2603'
>>> u'\N{SNOWMAN}'.encode('utf8', errors='surrogateescape')[::-1].decode('utf8', errors='surrogateescape')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
LookupError: unknown error handler name 'surrogateescape'
```